### PR TITLE
Configure github oauth in test cluster and add ESO resources for group sync operator and github oauth 

### PR DESCRIPTION
**Pull request by dystewart**
This PR addresses this issue: [https://github.com/OCP-on-NERC/operations/issues/125](https://github.com/OCP-on-NERC/operations/issues/125)

There are 2 separate commits:

1. Adds cluster oauth patch to the test cluster overlay to configure github IDP
2. Create secretstore and externalsecrets in the openshift-config namespace for github client secret and in the group-sync-operator namespace for github-group-sync secret

We need someone with admin access to the github OCP-on-NERC organization to setup a new github auth application to generate the github client secret which will be stored in vault. More info on that process [here](https://docs.openshift.com/container-platform/4.10/authentication/identity_providers/configuring-github-identity-provider.html). I've created the placeholder secret in vault now we just need the github auth application. @larsks could you assist with this step?

### CODE REVIEW

File: cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+    - name: github
+      mappingMethod: claim
+      type: GitHub
+      github:
+        clientID: 771ea98004d436c6e529
```
Comment (computate) [inline cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml:11]: Sounds like the `clientID` here is going to change if there is a new GitHub Auth application being created.

**Comment by dystewart** [comment]
Added [nerc-logs-metrics](https://github.com/orgs/OCP-on-NERC/teams/nerc-logs-metrics) team to test cluster oauth

File: cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        clientSecret:
+          name: github-client-secret
+        teams:
+          - ocp-on-nerc/nerc-ops
+          - nerc-logs-metrics
```
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml]: Based on the existing line, it looks like this should also be fully qualified (`ocp-on-nerc/nerc-logs-metrics`).

Comment (dystewart) [reply cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml]: Oops that's correct, careless copy/paste. Fixed

Comment (joachimweyl) [reply cluster-scope/overlays/nerc-ocp-test/oauths/cluster_patch.yaml]: @larsks with this resolved is this PR considered passing review by you?

**Comment by joachimweyl** [comment]
@computate when you get a chance please review this PR.

**Review by larsks — APPROVED** [review]

**Review by computate — APPROVED** [review]

**Pull request merged**